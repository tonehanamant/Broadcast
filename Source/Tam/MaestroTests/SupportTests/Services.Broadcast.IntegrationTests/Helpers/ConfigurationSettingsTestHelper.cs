using ApprovalTests;
using ApprovalTests.Reporters;
using Moq;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Services.Broadcast.IntegrationTests.Helpers
{

    public class ConfigurationSettingsTestHelper
    {
        protected ConfigurationSettingsHelper _GetConfigurationSettingsHelper()
        {
            return new ConfigurationSettingsHelper();
        }
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void LoadConfigItems()
        {
            var tc = _GetConfigurationSettingsHelper();
            var toVerify = tc._LoadConfigItems(BroadcastConstants.CONFIG_FILE_NAME);
            Approvals.Verify(IntegrationTestHelper.ConvertToJson(toVerify));
        }
        [Test]
        [TestCase("AABCacheExpirationSeconds", 300, 300)]
        [TestCase("ABC", 1800, 1800)]
        public void GetConfigValuesFromKeys<T>(string key, T defaultValue, T expectedResult)
        {
            var tc = _GetConfigurationSettingsHelper();
            var result = tc.GetConfigValueWithDefault(key, defaultValue);
            Assert.AreEqual(result, expectedResult);
        }
        [Test]
        [TestCase("UnitTestingPurposeKey", "Key 'UnitTestingPurposeKey' found but of incorrect type : test is not a valid value for Int32.")]
        [TestCase("ABC", "The key 'ABC' doesn't exist")]
        public void GetConfigValuesFromKeys_WhenNoValuePresent(string key, string expectedResult)
        {
            var tc = _GetConfigurationSettingsHelper();
            var exception = Assert.Throws<InvalidOperationException>(() => tc.GetConfigValue<int>(key));
            Assert.AreEqual(exception.Message, expectedResult);
        }
    }
}
