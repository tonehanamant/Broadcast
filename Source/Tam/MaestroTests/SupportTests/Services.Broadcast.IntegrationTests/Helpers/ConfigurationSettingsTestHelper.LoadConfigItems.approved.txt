{
  "ConnectionStrings:Broadcast": "Data Source=cpa-dev-sqldev.dev.cadent.tv\\maestro2_dev;Initial Catalog=broadcast_integration_staging;  Persist Security Info=True;user id=tamservice;pwd=KFqUjr+SjgugpL7h7yeJCg==; Asynchronous Processing=true",
  "ConnectionStrings:BroadcastForecast": "Data Source=cpa-dev-sqldev.dev.cadent.tv\\maestro2_dev;Initial Catalog=broadcast_forecast_integration_staging;  Persist Security Info=True;user id=tamservice;pwd=KFqUjr+SjgugpL7h7yeJCg==; Asynchronous Processing=true",
  "UnitTestingPurposeKey": "test",
  "AABCacheExpirationSeconds": 300,
  "BroadcastDayStart": 18000,
  "BroadcastMatchingBuffer": 120,
  "DefaultMarketCoverage": 0.8,
  "DefaultNtiConversionFactor": 0.2,
  "EfficiencyModelCpmGoal": 1,
  "EmailHost": "smtp.office365.com",
  "EmailNotificationsEnabled": true,
  "EmailPassword": "7TUCE+HAp3LDexQ6JIvaEA==",
  "EmailUsername": "broadcastsmtp@crossmw.com",
  "EnableBarterInventoryForPricingModel": "True",
  "EnableCampaignsLocking": true,
  "EnableOpenMarketInventoryForPricingModel": true,
  "EnableProprietaryOAndOInventoryForPricingModel": true,
  "FloorModelCpmGoal": 1,
  "InventoryProcessingGroupMonitorPauseSeconds": 300,
  "InventoryProcessingGroupTimeoutHours": 7,
  "InventoryProgramsEngineDeleteBatchSize": 1000,
  "InventoryProgramsEngineMaxDop": -1,
  "InventoryProgramsEngineParallelBatchSize": 10,
  "InventoryProgramsEngineParallelEnabled": false,
  "InventoryProgramsEngineSaveBatchSize": 1000,
  "InventoryRatingsParallelJobs": 3,
  "InventorySummaryCacheAbsoluteExpirationSeconds": 600,
  "MediaMonthCrunchCacheSlidingExpirationSeconds": 7200,
  "NumberOfFallbackQuartersForPricing": 8,
  "PlanPricingFloorPricingUrl": "https://datascience-dev.cadent.tv/broadcast-floor-pricing/v1/optimization",
  "PricingRequestLogAccessKeyId": "AKIAQJ5IV4IZZV35MPAM",
  "PricingRequestLogBucket": "s3://ds-api-logs",
  "PricingRequestLogBucketRegion": "us-east-1",
  "PricingRequestLogEncryptedAccessKey": "8WBxyR8JMnMGgdIk6I2aJkurXbm2Hgkwz1SV/hTsOoUtZ6UYnfBGQvCMaqNnrxjh",
  "RelativePathToMarketCoveragesFile": "~/App_Data/Market coverages.xlsx",
  "RunLongRunningJobsInWebServiceEnabled": false,
  "ScxGenerationJobIntervalSeconds": 300,
  "ScxGenerationParallelJobs": 3,
  "ThresholdInSecondsForProgramIntersectInPricing": 1800,
  "UseMaestroDaypartRepo": true,
  "AgencyAdvertiserBrandApiUrl": "TestAgencyAdvertiserBrandApiUrl",
  "BroadcastAppFolder": "c:\\temp\\Broadcast\\IntegrationTests",
  "BroadcastNTIUploadApiUrl": "",
  "BroadcastSharedFolder": "c:\\temp\\Broadcast\\IntegrationTests",
  "EmailFrom": "broadcast@crossmw.com",
  "EmailWhiteList": "SomeDev@cadent.tv",
  "InventoryProcessingNotificationEmails": "SomeDev@cadent.tv",
  "InventoryRatingsJobIntervalSeconds": 0,
  "LaunchDarklySdkKey": "SecretKeyForLaunchDarklySdkKey",
  "PlanPricingAllocationsEfficiencyModelUrl": "TestUrl",
  "PlanPricingAllocationsUrl": "http://datascience-dev.cadent.tv/broadcast-openmarket-allocations/v1/allocation",
  "ProgramSearchApiUrl": "",
  "ScxGenerationFolder": "c:\\temp\\Broadcast\\IntegrationTests\\ScxGeneration",
  "ReelIsciIngestNumberOfDays": 200,
  "BroadcastNotificationEmail": "broadcast@crossmw.com",
  "ReelIsciApi:ApiUrlBase": "TestUrlBase",
  "ReelIsciApi:EncryptedApiKey": "TestKey",
  "AttachmentMicroServiceApi:ApiBaseUrl": "https://internal-aws-test.cadent.tv/attachmentmicroservice/",
  "AttachmentMicroServiceApi:AppName": "Attachments",
  "AttachmentMicroServiceApi:ApplicationId": "D6A7417C-429A-45C4-AFD9-BDA5AEFAADE0"
}